webapp/src/features/adminAddIntersection/AdminAddIntersection.tsx

```diff
@@ -0,0 +1,219 @@
+import React, { useEffect, useState } from 'react'
+import { Form } from 'react-bootstrap'
+import { useForm } from 'react-hook-form'
+import { Multiselect } from 'react-widgets'
+import {
+  selectOrganizations,
+  selectSelectedOrganizations,
+  selectRsus,
+  selectSelectedRsus,
+  selectSubmitAttempt,
+
+  // actions
+  getIntersectionCreationData,
+  submitForm,
+  updateSelectedOrganizations,
+  updateSelectedRsus,
+} from './adminAddIntersectionSlice'
+import { useSelector, useDispatch } from 'react-redux'
+
+import '../adminIntersectionTab/Admin.css'
+import { AnyAction, ThunkDispatch } from '@reduxjs/toolkit'
+import { RootState } from '../../store'
+import toast from 'react-hot-toast'
+import { useNavigate } from 'react-router-dom'
+import Dialog from '@mui/material/Dialog'
+import { DialogActions, DialogContent, DialogTitle } from '@mui/material'
+
+export type AdminAddIntersectionForm = {
+  intersection_id: string
+  ref_pt: {
+    latitude: string
+    longitude: string
+  }
+  bbox?: {
+    latitude1: string
+    longitude1: string
+    latitude2: string
+    longitude2: string
+  }
+  intersection_name?: string
+  origin_ip?: string
+  organizations: string[]
+  rsus: string[]
+}
+
+const AdminAddIntersection = () => {
+  const dispatch: ThunkDispatch<RootState, void, AnyAction> = useDispatch()
+
+  const organizations = useSelector(selectOrganizations)
+  const selectedOrganizations = useSelector(selectSelectedOrganizations)
+  const rsus = useSelector(selectRsus)
+  const selectedRsus = useSelector(selectSelectedRsus)
+  const submitAttempt = useSelector(selectSubmitAttempt)
+
+  const [open, setOpen] = useState(true)
+  const navigate = useNavigate()
+
+  const notifySuccess = (message: string) => toast.success(message)
+  const notifyError = (message: string) => toast.error(message)
+
+  const handleFormSubmit = (data: AdminAddIntersectionForm) => {
+    dispatch(submitForm({ data, reset })).then((data: any) => {
+      data.payload.success
+        ? notifySuccess(data.payload.message)
+        : notifyError('Failed to add Intersection due to error: ' + data.payload.message)
+    })
+    setOpen(false)
+    navigate('/dashboard/admin/intersections')
+  }
+
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm<AdminAddIntersectionForm>()
+
+  useEffect(() => {
+    dispatch(getIntersectionCreationData())
+  }, [dispatch])
+
+  return (
+    <Dialog open={open}>
+      <DialogTitle>Add Intersection</DialogTitle>
+      <DialogContent>
+        <Form
+          id="add-intersection-form"
+          onSubmit={handleSubmit((data) => handleFormSubmit(data))}
+          style={{ fontFamily: 'Arial, Helvetica, sans-serif' }}
+        >
+          <Form.Group className="mb-3" controlId="ip">
+            <Form.Label>Intersection ID</Form.Label>
```

Comment:
Is the intersection ID necessary to allow the user to define it? Is this the value that gets used to match corresponding MAP and SPaT messages to?

Reply:
This is the intersection_number in the database, used to link up MAP and SPAT messages